cmd/download.go

```diff
@@ -185,7 +187,14 @@ func newDownload(flags *pflag.FlagSet, usrCfg *viper.Viper) (*download, error) {
 	}
 	defer res.Body.Close()
 
-	if err := json.NewDecoder(res.Body).Decode(&d.payload); err != nil {
+	if res.StatusCode < 200 || res.StatusCode > 299 {
+		return nil, decodedAPIError(res)
```

Comment:
Now we have the `decodedAPIError` twice, which seems like it shouldn't be necessary.

I'm sorry if I'm being dumb, here. I'll take some time after work to pull this down and play around with it to see if I understand it better.

Reply:
Ok, I played with this and I totally see what's going on.

Sorry for being slow!

Reply:
No problem, just glad I can help!